build/nuke/Build.Steps.cs

```diff
@@ -224,21 +229,52 @@ partial class Build
 
     private void RunUnitTests()
     {
-        Project[] unitTests = new[]
+        RunBootstrappingTests();
+
+        var unitTestProjects = new[]
         {
-                Solution.GetProject(Projects.Tests.ClrProfilerManagedLoaderTests)
+            Solution.GetProject(Projects.Tests.ClrProfilerManagedLoaderTests)
         };
 
         DotNetTest(config => config
             .SetConfiguration(BuildConfiguration)
             .SetTargetPlatformAnyCPU()
             .EnableNoRestore()
             .EnableNoBuild()
-            .CombineWith(unitTests, (s, project) => s
+            .CombineWith(unitTestProjects, (s, project) => s
                 .EnableTrxLogOutput(GetResultsDirectory(project))
                 .SetProjectFile(project)), degreeOfParallelism: 4);
     }
 
+    /// <summary>
+    /// Bootstrapping tests require every single test to be run in a separate process
+    /// so the tracer can be created from scratch for each of them.
+    /// </summary>
+    private void RunBootstrappingTests()
+    {
+        var project = Solution.GetProject(Projects.Tests.ClrProfilerManagedBootstrappingTests);
+
+        const string testPrefix = "OpenTelemetry.ClrProfiler.Managed.Bootstrapping.Tests.InstrumentationTests";
+        var testNames = new[] {
+            "Initialize_WithDisabledFlag_DoesNotCreateTracerProvider",
+            "Initialize_WithDefaultFlag_CreatesTracerProvider",
+            "Initialize_WithEnabledFlag_CreatesTracerProvider",
+            "Initialize_WithPreviouslyCreatedTracerProvider_WorksCorrectly"
+        }.Select(name => $"{testPrefix}.{name}");
```

Comment:
how about using reflection to get all the test names? otherwise, we can easily stop running some tests if e.g. we add more of them

Reply:
or at least give a comment in `test/OpenTelemetry.ClrProfiler.Managed.Bootstrapping.Tests/InstrumentationTests.cs` to make sure that the test names are synced